lib/reesmat.gi

```diff
@@ -77,15 +77,21 @@ end);
 InstallMethod(IsIdempotent, "for a Rees 0-matrix semigroup element",
 [IsReesZeroMatrixSemigroupElement], 
 function(x)
-  local R;
+  local matrix_entry, g;
 
-  R:=ReesMatrixSemigroupOfFamily(FamilyObj(x));
-  if IsGroup(UnderlyingSemigroup(R)) then 
-    # only for RZMS over groups!
-    return x![1]=0 or x![2]^-1=x![4][x![3]][x![1]];
-  else 
-    return x ^ 2 = x;
+  if x![1] = 0 then
+    # <x> is the 0 element of the family
+    return true;
   fi;
+
+  matrix_entry := x![4][x![3]][x![1]];
+
+  if matrix_entry = 0 then
+    return false;
+  fi;
+
+  g := UnderlyingElementOfReesZeroMatrixSemigroupElement(x);
+  return g * matrix_entry * g = g;
```

Comment:
Since I don't know how these things are stored internally, I cannot at all determine whether this change is valid. (E.g. an inversion happened in the old code which didn't happen before).
So I'll just trust that you know what you are doing :-).

Reply:
In case you are interested @fingolfin - if not, no need to read further!:

A Rees matrix semigroup element is a triple `(i, g, j)`, where `i` and `j` are indices (referring to a column and row of a matrix, respectively), `g` is an element of a semigroup, and the multiplication is
```
(i, g, j)(k, h, l) = (i, g * mat[j][k] * h, l)
```
where `mat` is the matrix associated to the Rees matrix semigroup. So an element is an idempotent whenever `g * mat[j][i] * g = g`. (The situation is slightly more complicated for Rees 0-matrix semigroups, but never mind).

For the special case that `g` is an element of a group (rather than of just a semigroup, in general), you can rearrange that equation to be `g = mat[j][i] ^ -1`; this is the condition that the old code checked. A better check would've been `g * mat[j][i] = One(g)`,
to avoid inversion.

The version I went with works in general (not just for groups), and doesn't seem to noticeably slow things down.